src/modules/musig/session_impl.h

```diff
@@ -0,0 +1,818 @@
+        secp256k1_ge_set_gej(&nonce_pts[i], &nonce_ptj);
+        secp256k1_declassify(ctx, &nonce_pts[i], sizeof(nonce_pts));
+        secp256k1_scalar_clear(&k[i]);
+    }
+    /* None of the nonce_pts will be infinity because k != 0 with overwhelming
+     * probability */
+    secp256k1_musig_pubnonce_save(pubnonce, nonce_pts);
+    return ret;
+}
+
+int secp256k1_musig_nonce_gen(const secp256k1_context* ctx, secp256k1_musig_secnonce *secnonce, secp256k1_musig_pubnonce *pubnonce, const unsigned char *session_secrand32, const unsigned char *seckey, const secp256k1_pubkey *pubkey, const unsigned char *msg32, const secp256k1_musig_keyagg_cache *keyagg_cache, const unsigned char *extra_input32) {
+    int ret = 1;
+    unsigned char acc = 0;
+    int i;
+
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(secnonce != NULL);
+    memset(secnonce, 0, sizeof(*secnonce));
+    ARG_CHECK(session_secrand32 != NULL);
+
+    /* Check in constant time that the session_secrand32 is not 0 as a
+     * defense-in-depth measure that may protect against a faulty RNG. */
+    for (i = 0; i < 32; i++) {
+        acc |= session_secrand32[i];
+    }
+    ret &= !!acc;
+    memset(&acc, 0, sizeof(acc));
+
+    /* We can declassify because branching on ret is only relevant when this
+     * function called with an invalid session_secrand32 argument */
+    secp256k1_declassify(ctx, &ret, sizeof(ret));
+    if (ret == 0) {
+        secp256k1_musig_secnonce_invalidate(ctx, secnonce, 1);
+        return 0;
+    }
+
+    return secp256k1_musig_nonce_gen_internal(ctx, secnonce, pubnonce, session_secrand32, seckey, pubkey, msg32, keyagg_cache, extra_input32);
+}
+
+int secp256k1_musig_nonce_gen_counter(const secp256k1_context* ctx, secp256k1_musig_secnonce *secnonce, secp256k1_musig_pubnonce *pubnonce, uint64_t nonrepeating_cnt, const secp256k1_keypair *keypair, const unsigned char *msg32, const secp256k1_musig_keyagg_cache *keyagg_cache, const unsigned char *extra_input32) {
+    unsigned char buf[32] = { 0 };
+    unsigned char seckey[32];
+    secp256k1_pubkey pubkey;
+    int ret;
+
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(secnonce != NULL);
+    memset(secnonce, 0, sizeof(*secnonce));
+    ARG_CHECK(keypair != NULL);
+
+    secp256k1_write_be64(buf, nonrepeating_cnt);
+    /* keypair_sec and keypair_pub do not fail if the arguments are not NULL */
+    ret = secp256k1_keypair_sec(ctx, seckey, keypair);
+    VERIFY_CHECK(ret);
+    ret = secp256k1_keypair_pub(ctx, &pubkey, keypair);
+    VERIFY_CHECK(ret);
+#ifndef VERIFY
+    (void) ret;
+#endif
+
+    if (!secp256k1_musig_nonce_gen_internal(ctx, secnonce, pubnonce, buf, seckey, &pubkey, msg32, keyagg_cache, extra_input32)) {
+        return 0;
+    }
+    memset(seckey, 0, sizeof(seckey));
+    return 1;
+}
+
+static int secp256k1_musig_sum_nonces(const secp256k1_context* ctx, secp256k1_gej *summed_nonces, const secp256k1_musig_pubnonce * const* pubnonces, size_t n_pubnonces) {
```

Comment:
```suggestion
static int secp256k1_musig_sum_pubnonces(const secp256k1_context* ctx, secp256k1_gej *summed_pubnonces, const secp256k1_musig_pubnonce * const* pubnonces, size_t n_pubnonces) {
```
perhaps a slightly better name

Reply:
done